.github/workflows/push_complement_image.yml

```
@@ -55,14 +55,20 @@ jobs:
           password: ${{ secrets.GITHUB_TOKEN }}
       - name: Work out labels for complement image
         id: meta
-        uses: docker/metadata-action@v1
+        uses: docker/metadata-action@v4
         with:
           images: ghcr.io/${{ github.repository }}/complement-synapse
+          tags: |
+            type=schedule,pattern=nightly,enable=${{ github.event_name == 'schedule'}}
+            type=raw,value=develop,enable=${{ github.event_name == 'schedule' || inputs.branch == 'develop' }}
+            type=raw,value=latest,enable=${{ github.event_name == 'push' || inputs.branch == 'master' }}
+            type=sha,format=long
       - name: Build complement image
```

```suggestion
      - name: Run scripts-dev/complement.sh to generate complement-synapse:latest image.
```

Looking back, it wasn't clear why we were tagging from `complement-synapse`

---

The corresponding line in complement.sh is

https://github.com/matrix-org/synapse/blob/69814eb2824daf846f869cb9579eb1008e61f8ad/scripts-dev/complement.sh#L113-L117

does this implicitly tag it as `:latest` locally? If so then I didn't need to remove the `:latest` from line `-66`

---

https://docs.docker.com/engine/reference/commandline/tag/#tag-an-image-referenced-by-name - with no tag after the image name, then it implicitly it gets forced to be latest.

---

( one reason i prefer to use :latest everywhere i can to be explicit)